lib/go-tc/serverchecks.go

```diff
@@ -35,36 +35,42 @@ type ServerchecksResponse struct {
 	Response []Servercheck `json:"response"`
 }
 
-// Servercheck is a single Servercheck struct for GET response.
-// swagger:model Servercheck
-type Servercheck struct {
+// CommonCheckFields is a structure containing all of the fields common to both
+// Serverchecks and GenericServerChecks.
+type CommonCheckFields struct {
 
-	// The Servercheck response data
-	//
-	// Admin state of the checked server
+	// AdminState is the server's status - called "AdminState" for legacy reasons.
 	AdminState string `json:"adminState"`
 
-	// Cache group the checked server belongs to
+	// CacheGroup is the name of the Cache Group to which the server belongs.
 	CacheGroup string `json:"cacheGroup"`
 
-	// ID number of the checked server
+	// ID is the integral, unique identifier of the server.
 	ID int `json:"id"`
 
-	// Hostname of the checked server
+	// HostName of the checked server.
 	HostName string `json:"hostName"`
 
-	// Reval pending flag for checked server
+	// RevalPending is a flag that indicates if revalidations are pending for the checked server.
 	RevalPending bool `json:"revalPending"`
 
-	// Profile name of checked server
+	// Profile is the name of the Profile used by the checked server.
 	Profile string `json:"profile"`
 
-	// Traffic Control type of the checked server
+	// Type is the name of the server's Type.
 	Type string `json:"type"`
 
-	// Update pending flag for the checked server
+	// UpdPending is a flag that indicates if updates are pending for the checked server.
 	UpdPending bool `json:"updPending"`
 
+}
+
+// Servercheck is a single Servercheck struct for GET response.
+// swagger:model Servercheck
```

Comment:
whats this swagger:model?

Reply:
That's just something that was there from when @dewrich was trying to get us to put our API documentation in swagger. I didn't remove it, but GoDocs *are* supposed to be complete sentences, so I can if it bothers you.